### PR TITLE
Calculate mailbox traversal times more accurately

### PR DESCRIPTION
**Pull request by larshesel**
This is a PoC to try to improve the mailbox traversal measurements.

Let me know what you think.

### CODE REVIEW

File: src/plumtree_broadcast.erl [hunk 1]
```diff
@@ -94,7 +102,13 @@
 
           %% Set of all known members. Used to determine
           %% which members have joined and left during a membership update
-          all_members   :: ordsets:ordset(nodename())
+          all_members   :: ordsets:ordset(nodename()),
+
+          %% State used to calculate the mailbox traversal time and to
+          %% back off the exchange of lazy sets to recover from an
+          %% overload situation.
+          mbox_traversal :: term()
```
Comment (dergraf) [inline src/plumtree_broadcast.erl]: `mbox_traversal :: #mbox_traversal{}` ?

File: src/plumtree_broadcast.erl [hunk 2]
```diff
@@ -307,6 +365,11 @@ handle_info({'DOWN', Ref, process, _Pid, _Reason}, State=#state{exchanges=Exchan
     Exchanges1 = lists:keydelete(Ref, 3, Exchanges),
     {noreply, State#state{exchanges=Exchanges1}}.
 
+next_mbox_traversal_msg(undefined) ->
+    {mbox_traversal_msg, 1, os:timestamp()};
+next_mbox_traversal_msg({mbox_traversal_msg, No, _}) ->
```
Comment (dergraf) [inline src/plumtree_broadcast.erl]: How and when is `No` used?

**Review by dergraf — COMMENTED** [review]
LGTM, if possible we should add a comment on top of the module that explains the keeping track of two tick messages to be able to calculate an improved tick interval.

**Comment by larshesel** [comment]
@dergraf I fixed your review comments and refactored the code a bit to make it easier to understand. Further I fixed a few dialyzer errors.

**Comment by larshesel** [comment]
Can we merge this - I'd like to get it into VerneMQ master so we can include it in the next release.

**Review by dergraf — COMMENTED** [review]
Last time I looked at it, it was ok.

**Pull request merged**